chall11/gbudau.c

```diff
@@ -0,0 +1,169 @@
+#include <stdlib.h>
+#include <string.h>
+#include <ctype.h>
+#define ERROR -1
+#define INFINITE_LOOP 1
+#define SUCCESS 0
+#define MAX(a,b) (((a)>(b))?(a):(b))
+
+static size_t	total_lines(const char *code)
+{
+	size_t	length;
+
+	length = 1;
+	for (; *code; code++)
+	{
+		if (*code == '\n')
+			length++;
+	}
+	return (length);
+}
+
+static void		split_lines(char *copy, char **split)
+{
+	char	*line;
+	size_t	i;
+
+	i = 0;
+	while ((line = strsep(&copy, "\n")) != NULL)
+	{
+		if (*line)
+		{
+			split[i] = line;
+			i++;
+		}
+	}
+	split[i] = NULL;
+}
+
+static char		*free_ret(char **unique, size_t length, int flag, char *chain)
+{
+	for (size_t i = 0; i < length; i++)
+		free(unique[i]);
+	free(unique);
+	if (flag == ERROR || flag == INFINITE_LOOP)
+	{
+		free(chain);
+		chain = NULL;
+	}
+	if (flag == INFINITE_LOOP)
+		return (strdup("Infinite loop !"));
+	return (chain);
+}
+
+static size_t	count_digits(const char *str)
+{
+	size_t	count;
+
+	count = 0;
+	for (int i = 0; isdigit(str[i]); i++)
+		count++;
+	return (count);
+}
+
+static char		*walk_goto(char *str)
+{
+	while (*str && isdigit(*str) == 0)
+		str++;
+	return str;
+}
+
+static ssize_t		goto_index(char **split, char *line)
+{
+	ssize_t	i = 0;
+	size_t	len1;
+	size_t	len2;
+
+	while (split[i] != NULL)
+	{
+		if (isdigit(*(split[i])))
+		{
+			len1 = count_digits(split[i]);
+			len2 = strlen(line);
+			if (strncmp(split[i], line, MAX(len1, len2)) == 0)
+				break;
```

Comment:
I hate `break` statements, I would have tried to approach this differently so I don't need to use it.

Reply:
The only ways i can think of writing it without a `break` statement is to add either a flag or to `return` instead of `break`, and both cases need writing redundant code.
I will keep it in mind nonetheless.